app/controllers/topics_controller.rb

```diff
@@ -939,6 +942,20 @@ def perform_show_response
     end
   end
 
+  def perform_hidden_topic_show_response(topic)
+    respond_to do |format|
+      format.html do
+        @topic_view = nil
+        render :show
+      end
+
+      format.json do
+        render_json_dump(HiddenTopicViewSerializer.new(topic, scope: guardian, root: nil))
```

Comment:
Is there a particular reason you didn't use `render_serialized` here? It's somewhat simpler.